velox/experimental/cudf/exec/CudfFilterProject.cpp

```diff
@@ -40,13 +41,64 @@ void debugPrintTree(
     debugPrintTree(input, indent + 2);
   }
 }
+
+bool checkAddIdentityProjection(
```

Comment:
This seems to be copied from `FilterProject.cpp`. Can it be extracted from both places to reduce code duplication? @Yuhta would it be ok if we did this for `FilterProject.cpp`?